src/dev/riscv/HiFive.py

```diff
@@ -129,19 +138,25 @@ def _off_chip_ranges(self):
             for dev in self._off_chip_devices()
         ]
 
-    def attachOnChipIO(self, bus):
-        """Attach on-chip IO devices, needs modification
-        to support DMA
-        """
-        for device in self._on_chip_devices():
-            device.pio = bus.mem_side_ports
-
-    def attachOffChipIO(self, bus):
-        """Attach off-chip IO devices, needs modification
-        to support DMA
-        """
-        for device in self._off_chip_devices():
-            device.pio = bus.mem_side_ports
+    def attachOnChipIO(self, bus, dma_ports=None):
+        for port in self._on_chip_ports():
+            if port.role == "GEM5 RESPONDER":
```

Comment:
```suggestion
            if isinstance(port, ResponsePort):
```

Reply:
Are you sure that it is better to use `isinstance()` here? 
As I understand it, `attachOnChipIO()/attachOffChipIO()` are called with ports which are instances of the `class PortRef` and to get back from them to the original `Request/ResponsePort` we would need to use something like:
```suggestion
            real_port = port.simobj._ports[port.name]
            if isinstance(real_port, ResponsePort) or isinstance(
                real_port, VectorResponsePort
            ):
```
In my opinion, version with just `port.role` looks better.

Reply:
You can leave it this way if you want but I'm always quite cautious at inferring types via comparison of strings. it only takes one typo to ruin your day